frontend/src/modules/EditZing/Components/Helpers.tsx

```diff
@@ -47,3 +121,99 @@ export async function saveSwapStudent(
       }
     )
 }
+
+// makes a div that contains a Gender: Female type beat thing for student grid
+export function makeItem(svg: JSX.Element, text: string) {
+  // case for where the text is a graduation year, since those fluctuate year by year, we don't want to access the json
+  if (String(Number(text)) === text) {
+    return (
+      <div>
+        {svg} {text}
+      </div>
+    )
+  }
+
+  return (
+    <div>
+      {svg} {text}
+    </div>
+  )
+}
+
+export function getChipIcon(category: string): JSX.Element {
+  let svg = otherSvg(14)
+  switch (category) {
+    case 'graduation':
+      svg = yearSvg(14)
+      break
+    case 'college':
+      svg = collegeSvg(14)
+      break
+    case 'identity':
+      svg = raceSvg(14)
+      break
+    case 'mode':
+      svg = modalitySvg(14)
+      break
+    case 'pronoun':
+      svg = genderSvg(14)
+      break
+    case 'location':
+      // i thought we got rid of if they're not on campus?
+      break
+    case 'start':
+      svg = timeSvg(14)
+      break
+    case 'time':
+      svg = workHabitsSvg(14)
+      break
+  }
+  return (
+    <div style={{ marginTop: '17px', marginLeft: '8px' }}>
+      <SvgIcon>{svg}</SvgIcon>
+    </div>
+  )
+}
+
+export function makeItems(student: Student, categoriesShown: CategoriesShown) {
+  let arrItems: JSX.Element[] = []
+  Object.keys(categoriesShown)
+    .sort() // sort bc keys return in random order
+    .forEach((key) => {
+      // guard for not student response attributes properties or not shown
+      if (
+        !['email', 'fullName', 'courseId'].includes(key) &&
+        categoriesShown[key]
+      ) {
```

Comment:
How about using `.filter` instead? Then `.map` can be used easily instead of pushing to the array